packages/server/src/sdk/app/rows/search/filters.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  FieldType,
+  RelationshipFieldMetadata,
+  SearchFilters,
+  Table,
+} from "@budibase/types"
+
+export function getRelationshipColumns(table: Table): {
+  name: string
+  definition: RelationshipFieldMetadata
+}[] {
+  return Object.entries(table.schema)
+    .filter(entry => entry[1].type === FieldType.LINK)
+    .map(entry => ({
+      name: entry[0],
+      definition: entry[1] as RelationshipFieldMetadata,
```

Comment:
Does this cast create a future danger that if we have subtypes of FieldType.LINK they may not be a RelationshipFieldMetadata?

Reply:
TLDR - I typed a long message about this, relationship fields already have some subtypes, this is the base type of all relationships already - however it would be nice if type narrowing could actually achieve this (with raw array filtering it can't without a cast of some format).

I've changed how this is done so that type narrowing achieves it, but it is a bit of a painful one with union types.